ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java

```diff
@@ -1064,6 +1084,13 @@ private void doRouterExceptionPipelineTest(FrontendTestRouter testRouter, String
               "routerExceptionPipelineTest");
           checkRouterExceptionPipeline(exceptionMsg, createRestRequest(restMethod, "/", headers, null));
           break;
+        case PUT:
+          testRouter.exceptionOpType = FrontendTestRouter.OpType.PutBlob;
```

Comment:
not required. Just need the `putFailureTest()` and the null inputs test (which you have added)